### PR TITLE
readme: add info about setting up App Catalog repo

### PR DESCRIPTION
**Pull request by kopiczko**
Fixes https://github.com/giantswarm/giantswarm/issues/6183.

### CODE REVIEW

**Comment by sslavic** [comment]
@kopiczko are you aware of https://github.com/giantswarm/giantswarm/blob/master/processes/appcatalog.md ?

**Comment by kopiczko** [comment]
> @kopiczko are you aware of https://github.com/giantswarm/giantswarm/blob/master/processes/appcatalog.md ?

No. How am I supposed to be if the review wasn't request from me?

**Comment by sslavic** [comment]
That's my bad, but we did agree that team batman will take over https://github.com/giantswarm/giantswarm/issues/6183

**Comment by kopiczko** [comment]
> That's my bad, sorry, but we did agree that team batman will take over [giantswarm/giantswarm#6183](https://github.com/giantswarm/giantswarm/issues/6183)

Why then I don't see any comment referencing the PR in the issue? Why the issue is still open?

**Comment by sslavic** [comment]
Not sure why https://github.com/giantswarm/giantswarm/issues/6183 stayed open after comment from @puja108, IMO it's safe to be closed. I guess like now, nobody (including you as assignee) even after getting that comment didn't really feel responsible to close it.

BTW Batman has https://github.com/giantswarm/giantswarm/issues/6078 to track (note checklist and checked item in the documentation section).

File: README.md [hunk 1]
```diff
@@ -74,9 +74,26 @@ Design and goals of the project:
...
+
+This job assumes that the App Catalog is defined in a GitHub repository inside
+giantswarm organization. E.g. when `app_catalog` parameter is set to
+`"control-plane-test-catalog"` the job will try to use catalog
+https://github.com/giantswarm/control-plane-test-catalog. All interactions with
+the App Catalog GitHub repository are done with [CatalogBot github
+user] credentials.
+
+Detailed instructions on how to set up App Catalog can be found here:
+https://github.com/giantswarm/giantswarm/blob/master/processes/appcatalog.md#setting-up-a-new-app-catalog.
```
Comment (kopiczko) [inline README.md]: @sslavic I linked here to the guide provided by you. I still want this to be merged. There are peaces of information that are implementation details not present in the other guide. And the other guide is too broad to included here.

**Pull request merged**